### PR TITLE
Pull Request Template

### LINKED ISSUE #698: Contribution: evaluate adding PR description templates
PRs should have a minimum description stating _WHAT_ has been changed and perhaps even _WHY_ the change was necessary. This will help reviewers greatly. Also, if there is a related issue, the PR should link to that issue. 

This issue deals with investigating whether those templates can be provided automatically, which seems to be the case according to the links below.

Once a suitable template has been created, the `contributing.md` file should document it as well.

- https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates
- https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/

### PR DESCRIPTION
**Pull request by paullatzelsperger**
adds an initial PR template 

Closes #698

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+## Linked Issue(s)
+
+Closes # <-- _insert Issue number if one exists_
+
+## Checklist
+
+- [ ] added appropriate tests?
+- [ ] performed checkstyle check locally?
+- [ ] added/updated Copyright headers?
```
Comment (florianrusch-zf) [inline .github/PULL_REQUEST_TEMPLATE.md:22]:
```suggestion
- [ ] added/updated copyright headers?
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+- [ ] documented public classes/methods?
+- [ ] added relevant documentation?
```
Comment (florianrusch-zf) [inline .github/PULL_REQUEST_TEMPLATE.md:24]:
```suggestion
- [ ] added/updated relevant documentation?
```

**Review by florianrusch-zf — APPROVED** [review]
Just two minor changes 😉

**Review by MoritzKeppler — APPROVED** [review]
fine for me, let's gather some experience and then adjust.

**Pull request merged**